utils/mantra_post_test_results

```diff
@@ -0,0 +1,92 @@
+#!/bin/sh
+
+set -e
+set -x
+
+usage() {
+    echo "Usage:"
+    echo "Usage:"
+    echo "  $0 --ensure <project-key> <build-name> <xml-file>"
+    echo "        Look up the build; if it does not exist, create it, then"
+    echo "        append <xml-file> test results to that build’s results."
+    echo
+    echo "  $0 <project-key> <build-name> <xml-file>"
+    echo "        Always create a *new* build named <build-name> and upload"
+    echo "        <xml-file> test results to it."
+    echo
+    echo "Project keys:"
+    echo "  client_accep_qemux86_64_uefi_grub"
+    echo "  client_accep_vexpress_qemu"
+    echo "  client_accep_qemux86_64_bios_grub"
+    echo "  client_accep_qemux86_64_bios_grub_gpt"
+    echo "  client_accep_vexpress_qemu_uboot_uefi_grub"
+    echo "  client_accep_vexpress_qemu_flash"
+    echo "  backend_integration_open_source"
+    echo "  backend_integration_enterprise"
+    echo "  full_integration"
+    echo "  client_accep_cross_platform"
+    exit 1
+}
+
+ensure=0
+if [ "$1" = "--ensure" ]; then
+    ensure=1
+    shift
+fi
+
+# we need at least 3 args now (after reading `--ensure`` flag): project-key, build-name, and results xml-file
+[ "$#" -lt 3 ] && usage
+
+project_key=$1 ; shift  # human readable project name
+build_name=$1  ; shift  # human-readable name (nightly-YYYY-MM-DD or pullreq-…)
+results_file=$1         # JUnit XML to upload
+
+# please note that there might be some gaps in the project_id numbers; this is related to some projects not being
+# used anymore but still results might be stored in Mantra. The important part is that the project_id numbers are unique.
+case "$project_key" in
+    client_accep_qemux86_64_uefi_grub)          project_id=1  ;;
+    client_accep_vexpress_qemu)                 project_id=2  ;;
+    client_accep_qemux86_64_bios_grub)          project_id=3  ;;
+    client_accep_qemux86_64_bios_grub_gpt)      project_id=4  ;;
+    client_accep_vexpress_qemu_uboot_uefi_grub) project_id=5  ;;
+    client_accep_vexpress_qemu_flash)           project_id=6  ;;
+    backend_integration_open_source)            project_id=7  ;;
+    backend_integration_enterprise)             project_id=8  ;;
+    full_integration)                           project_id=9  ;;
+    client_accep_cross_platform)                project_id=11 ;;
```

Comment:
Soon to be obsolete too, how are we planning to maintain this? Could we modify the interface of the script for the projectid to be passed as an input parameter? We risk callers overriding it, but as long as we have a table somewhere (ref my related comment on the help text) it should be fine.

Reply:
I think we should build the identifier of the upload from two components:
- project / repo name (get it from the CI global variables)
- manually set identifier (passed by user to better reflect what the test is doing)

I left it as-is for now for the backward compatibility and replace it when we will agree upon a name schema.